### PR TITLE
feat: add ButtonDropdown components state variable

### PR DESCRIPTION
**Pull request by iepn**
## Description

https://github.com/ethereum/ethereum-org-website/assets/57232813/3adcebe6-ba96-4af7-b48a-a6e2154dd1b5

Hello! Recently, I've been reading "The Design of Design" by Kenjiro Sano, which discusses symbols and associations, such as the association between a lamp's switch and a button. 

In light of this, I've noticed some issues with the ButtonDropdown component. 

### Preview
https://deploy-preview-12553--ethereumorg.netlify.app/en/staking/pools/#why-stake-with-a-pool

### CODE REVIEW

**Comment by wackerow** [comment]
At a glance, looks like you're recommending that we display the name of the section selected, correct? cc: @nloureiro @konopkja pinging either of you folks for design eyes

**Comment by iepn** [comment]
> At a glance, looks like you're recommending that we display the name of the section selected, correct? cc: @nloureiro @konopkja pinging either of you folks for design eyes


In other words, when you click on that section, it will display the title of that section. If nothing is done, it will display the default button title that has been set.

It's just like turning on lights. There are three lights, corresponding to three switches. When we click on the third switch, the third light will turn on.

**Comment by nloureiro** [comment]
I think this is a small improvement that needed to be made. It helps to understand that you switch the tab, and for small screens, it provides context.

From the UX perspective, it's a go.

I have other considerations to improve this mobile experience even further than switching/redesigning this dropdown; @iepn, should we open a "design issue" to explore future improvements?

**Review by nloureiro — APPROVED** [review]
From the design and UX it's a go

**Comment by iepn** [comment]
> I think this is a small improvement that needed to be made. It helps to understand that you switch the tab, and for small screens, it provides context.
> 
> From the UX perspective, it's a go.
> 
> I have other considerations to improve this mobile experience even further than switching/redesigning this dropdown; @iepn, should we open a "design issue" to explore future improvements?

I think what you've done is already very good. There are three lines in front of the button to indicate that there are many other sub-buttons under this button. There are even title prompts. From the indications, it's already very detailed.

If you have any new ideas, I'm happy to listen.

**Review by wackerow — APPROVED** [review]
Lgtm! Thanks again @iepn :fire:

**Comment by konopkja** [comment]
> At a glance, looks like you're recommending that we display the name of the section selected, correct? cc: @nloureiro @konopkja pinging either of you folks for design eyes

if showing the current state in the dropdown is the change discussed there then lgtm, wasnt sure what were all the changes made

**Comment by iepn** [comment]
> > At a glance, looks like you're recommending that we display the name of the section selected, correct? cc: @nloureiro @konopkja pinging either of you folks for design eyes
> 
> if showing the current state in the dropdown is the change discussed there then lgtm, wasnt sure what were all the changes made


Actually, his changes are super small, just four lines of code, appearing as `+21 −9`. In reality, `/StakingConsiderations/index.tsx` is a submission plan made to avoid https://github.com/ethereum/ethereum-org-website/pull/12535 from being merged (that is, the code on the main branch at the time). 

However,https://github.com/ethereum/ethereum-org-website/pull/12535 was merged before this PR was merged, so it's necessary to update the latest `index.tsx` code to avoid branch conflicts.

```diff
- import React, { MouseEvent } from "react"
+ import React, { MouseEvent, useState } from "react"
import { useTranslation } from "next-i18next"
import { MdMenu } from "react-icons/md"
import {
  Button,
  ButtonProps,
  Menu,
  MenuButton,
  MenuItem,
  MenuList,
} from "@chakra-ui/react"

// Utils
import { trackCustomEvent } from "@/lib/utils/matomo"

// Components
import { BaseLink } from "./Link"

export interface ListItem {
  text: string
  to?: string
  matomo?: {
    eventCategory: string
    eventAction: string
    eventName: string
  }
  callback?: (idx: number) => void
}

export interface List {
  text: string
  ariaLabel: string
  items: Array<ListItem>
}

export type ButtonDropdownProps = ButtonProps & {
  list: List
}

const ButtonDropdown = ({ list, ...rest }: ButtonDropdownProps) => {
  const { t } = useTranslation("common")
+  const [selectedItem, setSelectedItem] = useState(list.text)
  const handleClick = (
    e: MouseEvent<HTMLElement>,
    item: ListItem,
    idx: number
  ) => {
    const { matomo, callback } = item

    if (matomo) {
      trackCustomEvent(matomo)
    }

    if (callback) {
      e.preventDefault()
      callback(idx)
    }
+    setSelectedItem(item.text)
  }

  return (
    <Menu matchWidth>
      <MenuButton
        as={Button}
        leftIcon={<MdMenu />}
        variant="outline"
        _active={{ bg: "transparent" }}
        {...rest}
      >
-        {t(list.text)}
+        {t(selectedItem)}
      </MenuButton>
      <MenuList
        py={2}
        borderRadius="base"
        border="1px"
        borderColor="text"
        bg="dropdownBackground"
        zIndex="popover"
      >
        {list.items.map((item, idx) => {
          const { text, to } = item

          return to ? (
            <BaseLink
              key={idx}
              to={to!}
              isPartiallyActive={false}
              textDecor="none"
              color="text"
              fontWeight="normal"
              _hover={{ textDecor: "none", color: "primary.base" }}
              _focus={{ textDecor: "none", color: "primary.base" }}
            >
              <MenuItem
                as="span"
                onClick={(e) => handleClick(e, item, idx)}
                p={2}
                textAlign="center"
                justifyContent="center"
                bg="dropdownBackground"
                _hover={{
                  color: "primary.base",
                  bg: "dropdownBackgroundHover",
                }}
                _focus={{
                  color: "primary.base",
                  bg: "dropdownBackgroundHover",
                }}
              >
                {t(text)}
              </MenuItem>
            </BaseLink>
          ) : (
            <MenuItem key={idx} onClick={(e) => handleClick(e, item, idx)}>
              {t(text)}
            </MenuItem>
          )
        })}
      </MenuList>
    </Menu>
  )
}

export default ButtonDropdown
```

File: src/components/Staking/StakingConsiderations/index.tsx
Comment (wackerow) [inline src/components/Staking/StakingConsiderations/index.tsx:1]: @iepn Was there a reason for the changes in this file?

Comment (iepn) [reply src/components/Staking/StakingConsiderations/index.tsx:1]: > @iepn Was there a reason for the changes in this file?

This file was modified under the assumption that https://github.com/ethereum/ethereum-org-website/pull/12535 wouldn't be merged at that time. Therefore, I pasted the code from the original ethereum.org main branch. However, unexpectedly, it was merged, resulting in this situation

Comment (iepn) [reply src/components/Staking/StakingConsiderations/index.tsx:1]: I removed it in this latest commit. Could you please review if this was done correctly? Or does it have any impact on the main branch?

**Comment by wackerow** [comment]
@all-contributors please add @iepn for code

**Pull request merged**